src/pages/Wallet/NewAddressPage.tsx

```diff
@@ -63,39 +61,33 @@ const NewAddressPage = () => {
 
   const onGenerateClick = () => {
     if (newAddressData && newAddressGroup !== undefined) {
-      saveNewAddress({
-        hash: newAddressData.address,
-        publicKey: newAddressData.publicKey,
-        privateKey: newAddressData.privateKey,
-        group: newAddressGroup,
-        index: newAddressData.addressIndex,
-        settings: {
-          isMain: isMainAddress,
-          label: addressLabel.title,
-          color: addressLabel.color
-        },
-        transactions: {}
-      })
-      if (isMainAddress && currentMainAddress && currentMainAddress.index !== newAddressData.addressIndex) {
-        saveNewAddress({
-          ...currentMainAddress,
-          settings: {
-            ...currentMainAddress.settings,
-            isMain: false
+      saveNewAddress(
```

Comment:
Shouldn't we await this and stop the flow if the fetch fails?

Reply:
I think the user should be able to generate and locally save new addresses, even if the internet connection fails (fetch fails). WDYT?

Reply:
Would the "address data" be fetched properly (== clean state) once the internet connection is back? If yes, then yes, all good!

Reply:
It would be fetched property when the internet connection is back and one of the following happens:
- the user clicks the "refresh" button
- the wallet gets locked and unlocked

Basically, there is no implemented mechanism to set an interval to keep fetching address data if a request fails.